services/api/src/app.py

```diff
@@ -130,26 +133,28 @@ async def request(
     return response
 
 
-async def _blocking_response(response: ResponseModel):
-    logger.info(f"Responding to SID: `{response.session_id}`:")
-    response.log(logger)
+@sm.on("connect")
+async def connect(session_id: str, environ: Dict):
+    params = environ.get("QUERY_STRING")
+    params = dict(x.split("=") for x in params.split("&"))
+    
+    if "job_id" in params:
 
-    await sm.emit(
-        "blocking_response",
-        data=response.model_dump(exclude_defaults=True, exclude_none=True),
-        to=response.session_id,
-    )
+        await sm.enter_room(session_id, params["job_id"])
 
 
-@app.post("/blocking_response")
-async def blocking_response(response: BackendResponseModel):
-    """Endpoint to have server respond to sid.
+@sm.on("blocking_response")
+async def blocking_response(session_id: str, client_session_id: str, data: Any):
+
+    await sm.emit("blocking_response", data=data, to=client_session_id)
 
-    Args:
-        id (str): _description_
-    """
 
-    await _blocking_response(response)
+@sm.on("stream_upload")
+async def stream_upload(session_id: str, value: Dict):
```

Comment:
Doesn't seem like `session_id` is used here (along with many of the `sm.on` decorators), is this just because the session manager always passes `session_id` as the first arg?